lib/FusionInventory/Agent/Task/Inventory/Generic/Storages/HP.pm

```diff
@@ -5,159 +5,20 @@ use warnings;
 
 use parent 'FusionInventory::Agent::Task::Inventory::Module';
 
-use English qw(-no_match_vars);
-use UNIVERSAL::require;
-
 use FusionInventory::Agent::Tools;
-# Tested on 2.6.* kernels
-#
-# Cards tested :
-#
-# Smart Array E200
-#
-# HP Array Configuration Utility CLI 7.85-18.0
-
-# This speeds up hpacucli startup by skipping non-local (iSCSI, Fibre) storages.
-# See https://support.hpe.com/hpsc/doc/public/display?docId=emr_na-c03696601
-$ENV{INFOMGR_BYPASS_NONSA} = "1";
-
-sub _getHpacuacliFromWinRegistry {
-
-    my $Registry;
-    Win32::TieRegistry->require();
-    Win32::TieRegistry->import(
-        Delimiter   => '/',
-        ArrayValues => 0,
-        TiedRef     => \$Registry,
-    );
-
-    my $machKey = $Registry->Open('LMachine', {
-        Access => Win32::TieRegistry::KEY_READ(),
-    }) or die "Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR";
-
-    my $uninstallValues =
-        $machKey->{'SOFTWARE/Microsoft/Windows/CurrentVersion/Uninstall/HP ACUCLI'};
-    return unless $uninstallValues;
-
-    my $uninstallString = $uninstallValues->{'/UninstallString'};
-    return unless $uninstallString;
-
-    return unless $uninstallString =~ /(.*\\)hpuninst\.exe/;
-    my $hpacuacliPath = $1 . 'bin\\hpacucli.exe';
-    return unless -f $hpacuacliPath;
+use FusionInventory::Agent::Tools::Storages::HP;
 
-    return $hpacuacliPath;
-}
+our $runMeIfTheseChecksFailed = ['FusionInventory::Agent::Task::Inventory::Generic::Storages::HpWithSmartctl'];
 
 sub isEnabled {
-    return
-        canRun('hpacucli') ||
-        ($OSNAME eq 'MSWin32' && _getHpacuacliFromWinRegistry());
+    return canRun('hpacucli');
```

Comment:
To answer your last question about Win32::Storages::HP, keep it but to avoid the hypothetical case where `hpacucli` could be found on MSWin32, update this to:
```suggestion
    # MSWin32 has its Win32::Storages::HP dedicated module
    return canRun('hpacucli') && $OSNAME ne 'MsWin32';
```